scripts/configure.py

```diff
@@ -0,0 +1,136 @@
+    target: string
+        the target local path of the downloaded file
+    retries: int
+        the max number of retries allowed for urlretrieve
+    """
+    assert retries >= 1, "Number of retries should be at least 1"
+
+    retry = 0
+    while retry < retries:
+        try:
+            urllib.urlretrieve(url, target)
+            return
+        except Exception as e:
+            retry += 1
+            if retry < retries: 
+                time.sleep(2 ** retries * base_retry_interval)
+            else:
+                raise e
+
+
+def check_existence(filename, path):
+    """Check the existence of filename in path
+
+    Parameters
+    ----------
+    filename: string
+        name of file or dir to verify existence
+    path: string
+        path to verify the existence
+
+    Returns
+    -------
+    bool: Returns True if exist, False otherwise
+    
+    Examples
+    ----------
+    >>> check_existence('mxnet/', lambda_function_path)
+    >>> check_existence('MAR-INF/requirements.txt', lambda_function_path)
+    """
+    return os.path.exists(os.path.join(path, filename))
+
+
+def configure(model_archive, lambda_function_path, model_bucket=None):
+    """Configure Lambda Function to consume Model Archive
+    
+    Parameters
+    ----------
+    model_archive: string
+        url or path to the local model archive file
+    model_bucket: string
+        optional S3 bucket name to upload local model archive file
+    """
+    # check weither it is url
+    is_url = "://" in model_archive
+    url_mar = None
+    if is_url==False and model_bucket==None:
+        raise Exception("Please specify S3 bucket for model storage")
+    # create a tmp path
+    dirpath = tempfile.mkdtemp()
+    # download if is url
+    if is_url:
+        download_url(model_archive, os.path.join(dirpath, "tmp.mar"))
+        url_mar = model_archive
+        model_archive = os.path.join(dirpath, "tmp.mar")
+    # unzip
+    status = subprocess.call(" ".join(["unzip", model_archive, "-d", dirpath]), shell=True)
+    if status != 0:
+        raise Exception("Failed to unzip model archive. Interrupt.")
+    # install requirements
+    if check_existence('MAR-INF/requirements.txt', dirpath):
+        do_install(os.path.join(dirpath, 'MAR-INF/requirements.txt'), requirement=True, target=lambda_function_path)
+    if check_existence('mxnet/', dirpath) == False and check_existence('mxnet/', lambda_function_path) == False:
+        # rollback numpy to 1.13 (avoid problems with the latest version)
+        do_install('numpy==1.13.3', requirement=False, target=lambda_function_path)
+        do_install('mxnet', requirement=False, target=lambda_function_path)
+    # remove temp path
+    shutil.rmtree(dirpath)
+    if not is_url:
+        # upload model_archive
+        status = subprocess.call(" ".join(["aws s3 cp", model_archive, "s3://", model_bucket, "--grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers"]), shell=True)
```

Comment:
please mention somewhere in the documentation that a pre-requisite for this script is to have aws cli installed.

Reply:
@nswamy @apeforest  Progress is actually nicely logged with aws s3 cp subprocess.